### PR TITLE
[JBPM-8943] business central cannot show process details when configured with multiple kie-servers through smart router.

### PR DESCRIPTION
**Pull request by elguardian**
Added logic to avoid deep merge when the call is single result

### CODE REVIEW

**Comment by elguardian** [comment]
@mswiderski @MarianMacik not very convince about this. 

Several paths:
1) build a list of single result calls to kie-server
2) build a list of list result calls to kie-server

3) build dynamically of list result calls (search for implementors of ItemList; needs a dependency) -> not sure we want that

ideas are welcome

**Comment by mswiderski** [comment]
@elguardian how about flagging responses with some headers about type of the response - list or single item. If not given assuming it's list.... wdyt?

**Comment by elguardian** [comment]
@mswiderski works for me

**Comment by elguardian** [comment]
@mswiderski changed. this approach seems cleaner but the problem is we need to go one by one call. check and tell me wdyt

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/utils/MediaTypeUtil.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package org.kie.server.router.utils;
```
Comment (mswiderski) [inline kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/utils/MediaTypeUtil.java]: missing license header

**Comment by MarianMacik** [comment]
jenkins retest this

**Comment by elguardian** [comment]
sure, will change the implementation.

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-common/src/main/java/org/kie/server/remote/rest/common/util/RestUtils.java [hunk 2]
```diff
@@ -94,12 +97,23 @@ public static Response createResponse(Object responseObj, Variant v, javax.ws.rs
...
+        return getVariant(headers, null);
+    }
+
+    public static Variant getVariant(HttpHeaders headers, Object obj) {
         Variant v = RestEasy960Util.getVariant(headers);
         if( v == null ) {
             v = Variant.mediaTypes(getMediaType(headers)).add().build().get(0);
         }
+
+        if (obj instanceof ItemList || (obj instanceof ServiceResponse && ((ServiceResponse) obj).getResult() instanceof ItemList)) {
```
Comment (elguardian) [inline kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-common/src/main/java/org/kie/server/remote/rest/common/util/RestUtils.java]: @mswiderski I calculated the objects being ItemList as the only ones aggregatable. I don't see if others must be in that regard.

**Comment by elguardian** [comment]
added @Aggregatable annotation to define whether an object response could be aggregate or not.

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by elguardian** [comment]
@gom

> It looks good to me.
> 
> Just one question and one consideration @elguardian:
> 
> * Is ProcessInstance the unique model for adding Aggregatable(false) annotation? Are there no more entities that should include this annotation to avoid being aggregated?
> * Regarding aggregated elements: for these cases, requested limit applies to each kie-server behind the router, returning this one the sum of all (requested limit is not maintained, perhaps this should be documented anyway).

regarding your first bullet. Only this one is the goal of this bug.
regarding your second bullet. That is not the goal of the fix (prior condition)

**Comment by elguardian** [comment]
@kiegroup/gatekeepers  can anybody take care of the merge ?

**Pull request merged**